src/common/src/catalog/mod.rs

```diff
@@ -295,24 +294,17 @@ impl From<&TableOption> for risingwave_pb::hummock::TableOption {
 }
 
 impl TableOption {
-    pub fn build_table_option(table_properties: &HashMap<String, String>) -> Self {
+    pub fn new(retention_seconds: Option<u32>) -> Self {
         // now we only support ttl for TableOption
-        let mut result = TableOption::default();
-        if let Some(ttl_string) = table_properties.get(hummock::PROPERTIES_RETENTION_SECOND_KEY) {
-            match ttl_string.trim().parse::<u32>() {
-                Ok(retention_seconds_u32) => result.retention_seconds = Some(retention_seconds_u32),
-                Err(e) => {
-                    tracing::info!(
-                        error = %e.as_report(),
-                        "build_table_option parse option ttl_string {}",
-                        ttl_string,
-                    );
-                    result.retention_seconds = None;
+        TableOption {
+            retention_seconds: retention_seconds.and_then(|ttl| {
```

Comment:
Is it essentially `Option<NonZeroU32>`?

Reply:
change it to `Option<u32>` now, because most of the places use the prost's struct directly.